worker/worker.go

```diff
@@ -1144,7 +1144,7 @@ func (w *Worker) headObject(ctx context.Context, bucket, key string, onlyMetadat
 	})
 	if err != nil {
 		return nil, api.Object{}, fmt.Errorf("couldn't fetch object: %w", err)
-	} else if res.Object == nil {
+	} else if res.Object == nil && !onlyMetadata {
```

Comment:
@ChrisSchinnerl is there a particular reason why we want to return this error? It seems like we can drop it all together.

Reply:
It's probably a leftover.